### ISSUE TITLE
[discussion]: Tertiary navigation options

### ISSUE THREAD

**Issue by vpicone** [carbon-design-system/gatsby-theme-carbon#780]
We're looking for teams and site's that would benefit from tertiary level navigation. A tertiary level of navigation would mean a layer beyond the top level menu items (primary), and sub-menu items (secondary).

[Tabs](https://gatsby-theme-carbon.now.sh/guides/navigation/tabs) are already available as a form of tertiary navigation. However, they're not super discoverable and can result in a confusing information architecture.

Some options we thought were interesting are 
- [Polaris](https://polaris.shopify.com/content/voice-and-tone), who divide their content up by a top level category. This results in a more consistent top-down/inverted implementation of Tabs. Other examples of this pattern include:
	- [Google's material design system](https://material.io/)
	- [Microsoft's Fabric design system](https://developer.microsoft.com/en-us/fabric#/styles/web)
	- [ant.design](https://ant.design/)

- [Atlassian Design](https://www.atlassian.design/guidelines/designPrinciples/design-principles) – this options results in switching out the left nav entirely. Other instances of the left nav are accessible through a dropdown menu.

**Comment by theiliad** [comment]
![image](https://user-images.githubusercontent.com/14989804/76993945-0e6d3a80-6924-11ea-847f-a915ede21019.png)

I’ll add a bit more context to the `problem statement` here, and @jeanservaas might chime in as well.

Currently within the Carbon website for instance you’ve got 4-5 key sections to the IA at the highest level (`tutorials`, `patterns`, `guidelines`, `demos` etc…)

Now within the `Data Visualization` nav item you’ve got the same architecture setup, however there are no distinguishing elements to differentiate the sub nav items in there.
We’ve discussed this in depth with @jeanservaas, @sadekbazaraa and @mjabbink and I believe it’s pretty clear that tabs won’t be the ideal option.
I’m open to continue exploring other alternatives however tertiary navigation seems to be the most viable solution currently

**Comment by vpicone** [comment]
@theiliad My initial thought is that If Data Vis requires an AI equally as complex as Carbon, then it might be better served on its own site. Doubling down on the complexity by increasing nav depth seems like a cop out. 

With a third level we would have:
1. Top level
2. top level tab
3. Second level
4. second level tab
5. third level
6. third level tab

In some propositions (Brand), you get a whole new nav bar. This means EIGHT layers when you consider tabbed pages. 

Getting people to organize their data effectively with 6-8 possible IA layers is a big ask.

**Comment by theiliad** [comment]
In my perspective the IA value of tabs is not equal to sidenav elements.

Tabs aren't the most ideal way to handle navigation of your content and rather offer a way to hide & show various smaller pieces within 1 page (e.g. tabs should be used within a single page rather than offering a way to navigate through different pages)

Therefore I personally wouldn't consider tabs as a navigational layer.

I believe @sadekbazaraa has gone through multiple explorations and iterations of IA with Carbon. Sadek feel free to chime in on any of your findings.

**Comment by tay1orjones** [comment]
The example from Polaris would be beneficial for us. Our site is emerging into two primary use cases:
1. PAL Documentation (design and dev) - effectively the same use case as the Carbon website.
2. An information hub for the internal product design teams. Things like
   - Designer onboarding info
   - Product team contacts/roles/etc
   - Design Research process documentation
   - ...basically things you'd typically find previously in a connections wiki or similar

It would be incredibly helpful to separate these two at the topmost level while retaining the existing levels of navigation (left nav groups -> tabbed pages -> anchorlinks).

Currently it looks like we'll be aiming to hack a dividing line in our left nav to try and separate these which is far from ideal. The screens below aren't an exact representation, but you get the drift:
| existing | with new dividing line |
| - | - |
| ![image](https://user-images.githubusercontent.com/3360588/78926846-6ac70400-7a63-11ea-86fd-64422b1595a3.png) | ![image](https://user-images.githubusercontent.com/3360588/78926803-571b9d80-7a63-11ea-80ea-2928ded49f03.png) |

**Comment by sadekbazaraa** [comment]
@tay1orjones  @vpicone I would recommend using the vertical dividers that we're already using when considering this as an option. The same style as what you have above "Storybook".

**Comment by tay1orjones** [comment]
@sadekbazaraa Yes, that is the planned visual treatment. My screenshot above is a mockup quickly done via dev tools to show the general intention. The specific implementation would have the dividing line with the same treatment as the one above the Storybook link.
